### ISSUE TITLE
Question about stage 2 training data

### ISSUE THREAD

**Issue by bruceisme** [hanxunyu/DepthVLM#10]
Did you incorporate any general VQA datasets during Stage 2 training? Which ones should be used in stage2?
If training solely with the datasets presented in the paper, the resulting model will only generate fixed responses.

**Comment by bruceisme** [comment]
We trained a qwen3-vl-4b only on the nuscenes datasets

**Comment by hanxunyu** [comment]
Hi @bruceisme, we do not use any general VQA datasets for training.

For the resulting model, if the input prompt is the training prompt:

> “Given this image, estimate the metric depth (in meters, rounded to two decimal places) for every pixel of the image.”

the model will return the fixed response:

> “OK, I will estimate the depth map of this image.”

This response is only used to indicate the task mode for metric depth estimation.

If you test the model with any other prompts, such as:

> “What is the shortest distance between the front-left vehicle and the yellow detour sign on the right?”

or

> “What is the depth order of the trash can, sink, and electric kettle from nearest to farthest?”

the resulting model will produce normal and diverse responses, similar to a general VLM.

Hope this helps!

**Comment by bruceisme** [comment]
你好，我参照您的论文与代码的数据处理、stage1与stage2训练脚本，没有改动任何训练相关参数，只使用nuscenes去训练Qwen3vl-4B-instruct，得到的二阶段模型在生成深度图时效果显著。但是在进行正常VQA询问时，二阶段模型基本上只能输出类似“OK, I will estimate the depth map of this image.”这类的回答，我们只是单纯的使用了类似“物体A比物体B近还是远”之类的问题，并且训练得到的模型在VLMEvalKit上测试得到的精度也远低于原版模型(我们在mmbench_EN基准的输出上观察到大量的“OK, I will estimate ..."的输出)。是我们这边的训练操作有可能忽略了哪些必要设置吗？或者您有什么建议能够帮助我们复现论文中的效果吗？

**Comment by hanxunyu** [comment]
> “我们只是单纯的使用了类似“物体A比物体B近还是远”之类的问题”

It's very strange. Did you use any additional VQA data during Stage 2 training? We did not construct any additional VQA data. Both Stage 1 and Stage 2 were trained only on data related to metric depth estimation, to avoid degrading the VLM’s general VQA capabilities.

**Comment by bruceisme** [comment]
What I mean is that the test data queries we ran on the stage2 model are all tasks of judging whether Object A is closer or farther than Object B. However, the model only responds with "OK, I will estimate ...". 
We followed the exact training procedures from your open-source code. The only difference is the smaller training dataset—we only used nuScenes.
Actually, we think this is logically reasonable. For the Stage 2 dataset built from the DepthVLM repo, all queries and answers are fixed. If we train only on this data, the model’s outputs will be heavily affected. Our reproduction experiments confirm this. That’s why we are curious: did you use extra datasets during training, or could there be incorrect parameter settings somewhere in the codebase?

**Comment by bruceisme** [comment]
We are pleased to see that the depth map testing scripts within the codebase are fairly comprehensive. Furthermore, could you let us know if you can release the code for evaluating general VQA performance?

**Comment by hanxunyu** [comment]
The evaluation of general VQA performance is conducted using [LMMs-Eval](https://github.com/EvolvingLMMs-Lab/lmms-eval).

In our experiments, the model only produces a fixed response when the question is exactly the same as the one used during training. When asked different questions, the fine-tuned model generates diverse responses. I may need some time to check the parameters and training settings in the codebase.
